web/src/message_fetch.ts

```diff
@@ -235,6 +236,7 @@ function handle_operators_supporting_id_based_api(data: {
     const operators_supporting_id = new Set([
         "id",
         "stream",
+        "channel",
```

Comment:
I'm pretty sure that the web app is only going to have "channel" here. But if we want to keep this aligned with the same `operators_supporting_id` in `zerver/lib/narrow.py`, then these changes make sense. If we're fine with those two sets being different, then we can just have "channel" here and check for that one string below.

Reply:
I think it's fine for these to be different from the backend if we expect it'll have been canonicalized to `channel` before getting here. We could consider changing the server-side implementation to canonicalize early too.

Reply:
I put up what I think is a better fix for this in #31934, which refactors `handle_operators_support_id_based_api` and `loading_messages`. Since this is somewhat tangential to the changes to generate URLs with the "channel" operator, I'm going to drop this commit from this PR.